invenio_records_resources/records/systemfields/pid.py

```diff
@@ -180,3 +184,83 @@ def dump_obj(field, record, pid):
 
         # Set ID on desired dictionary key.
         field.set_dictkey(record, pid.pid_value)
+
+
+class ModelPIDFieldContext(PIDFieldContext):
+    """Context for ModelPIDField.
+
+    This class implements the class-level methods available on a PIDField. I.e.
+    when you access the field through the class, for instance:
+
+    .. code-block:: python
+
+        Record.pid.resolve('...')
+        Record.pid.session_merge(record)
+    """
+
+    def resolve(self, pid_value, registered_only=True):
+        """Resolve identifier."""
+        resolver = self.field._resolver_cls(
+            self._record_cls, self.field.model_field_name
+        )
+        pid, record = resolver.resolve(pid_value)
+        self.field._set_cache(record, pid)
+
+        return record
+
+    def create(self, record):
+        """Method to create a new persistent identifier for the record."""
+        # pop from metadata
+        pid_value = record.pop(self.field.model_field_name)
+        self.field._provider.create(
+            pid_value, record, self.field.model_field_name
+        )
+        # setattr(record, self.field.model_field_name, pid)
+        # record.pid_status = PIDStatus.REGISTERED  # trigger __set__
```

Comment:
I guess these can be removed (I see they're handled in the provider)

Reply:
Yup, sorry. Leftovers...